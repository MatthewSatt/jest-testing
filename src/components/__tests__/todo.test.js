import {render, screen, cleanup} from '@testing-library/react'
import Todo from "../../components/todo"
import renderer from 'react-test-renderer'
afterEach(() => {
    cleanup()
})

test('should render the to do component', () => {
    const todo = { id: 1, title:'clean room', completed: false}
    render(<Todo todo={todo}/>);
    const toDoElement = screen.getByTestId(`todo-1`);
    expect(toDoElement).toBeInTheDocument();
    expect(toDoElement).toHaveTextContent("clean room")
    expect(toDoElement).not.toContainHTML("<strike>")
})


test('should render the to do component', () => {
    const todo = { id: 5, title:'whatever', completed: true}
    render(<Todo todo={todo}/>);
    const toDoElement = screen.getByTestId(`todo-5`);
    expect(toDoElement).toBeInTheDocument();
    expect(toDoElement).toHaveTextContent("whatever")
    expect(toDoElement).toContainHTML("strike")
})

test("matches snapchat", () => {
    const todo = { id: 1, title:'clean room', completed: false}
    const tree = renderer.create(<Todo todo={todo} />).toJSON()
    expect(tree).toMatchSnapshot()
})
