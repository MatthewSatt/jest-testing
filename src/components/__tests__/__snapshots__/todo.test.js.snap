// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapchat 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    clean room
  </h1>
</div>
`;
